File: tests/nim/context.nim
import
  py2nim_helpers

type
  VMError* = object of Exception
type
  A* = object
    a*: int

proc enter*(self: var A): void =
  self.a = 2

proc exit*(self: var A; excType: ref Exception; excValue: ref Exception; traceback: string): void =
  if notexcValue.isNil() and excValue of VMError:
    echo "ERROR"
  else:
    self.a = -2
  
var a = A(a: 0)
try:
  with a,
    raise newException(VMError, "")
except :
  echo 0

File: tests/python/context.py
class VMError(Exception):
    pass


class A:

    def __init__(self, a):
        self.a = a

    def __enter__(self):
        self.a = 2

    def __exit__(self, exc_type, exc_value, traceback):
        if exc_value and isinstance(exc_value, VMError):
            print('ERROR')
        else:
            self.a = -2

a = A(0)
try:
    with a:
        raise VMError('')
except:
    print(0)
